src/Helper/Customer/Utility.php

```diff
@@ -184,9 +192,41 @@ protected function setAddresses($shopgateCustomer, $magentoCustomer)
             $shopgateAddress->setZipcode($mageAddress->getPostcode());
             $shopgateAddress->setCountry($mageAddress->getCountryId());
             $shopgateAddress->setState($this->regions->getIsoStateByMagentoRegion($mageAddress));
-
+            $shopgateAddress->setCustomFields(
+                $this->getShopgateCustomFields($mageAddress, self::ADDRESS_CUSTOM_FIELD_WHITELIST)
+            );
             $addresses[] = $shopgateAddress;
         }
         $shopgateCustomer->setAddresses($addresses);
     }
+
+    /**
+     * @param CustomAttributesDataInterface $mageAddress
+     * @param string[]                      $customFieldKeys
+     *
+     * @return ShopgateOrderCustomField[]
+     */
+    protected function getShopgateCustomFields($mageAddress, $customFieldKeys)
```

Comment:
$mageAddress could be $mageCustomer too, so please refactor to something more general.
What is the first common class, both objects inherit from?

Reply:
changed name of $mageAddress parameter. I already found the common ancestor shared by $mageAddress and $mageCustomer. I just didn't think to change the param name.